### ISSUE TITLE
Zoom buttons harmless crash

### ISSUE THREAD

**Issue by monsieurtanuki** [osmdroid/osmdroid#736]
## Issue Type
[x] Bug (but an invisible and harmless one)

## Description and/or steps/code to reproduce the problem
* open "OSMDroid Sample map (Start here)"
* tap on the map - the zoom buttons will appear
* tilt your smartphone (landscape/portrait) while the zoom buttons are visible - the app won't crash at all, but an exception crash will appear in the logs, something like
````
E/WindowManager: android.view.WindowLeaked: Activity org.osmdroid.ExtraSamplesActivity has leaked window android.widget.ZoomButtonsController$Container{2f3b6972 V.E..... ........ 0,0-480,73} that was originally added here
at android.view.ViewRootImpl.<init>(ViewRootImpl.java:371)
at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:261)
at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
at android.widget.ZoomButtonsController.setVisible(ZoomButtonsController.java:370)
at org.osmdroid.views.MapView$MapViewGestureDetectorListener.onDown(MapView.java:1303)
at android.view.GestureDetector.onTouchEvent(GestureDetector.java:550)
at org.osmdroid.views.MapView.dispatchTouchEvent(MapView.java:960)
````

**Comment by spyhunter99** [comment]
I've seen this before,  I think there is a ticket already.  The built in zoom controls are an android thing,  we should just get rid of it and roll our own solution

**Comment by monsieurtanuki** [comment]
We could use the zoom buttons that are already used in demo # 2 "OSMapView with Minimap, ZoomControls, Animations, Scale Bar and MyLocationOverlay" (Activity `SampleExtensive`).
Then the questions are:
* where do we display them? (same place as current standard zoom control I guess)
* do we play a fade in / fade out animation? (I hope not)

**Comment by spyhunter99** [comment]
well we could make a view that is programmatically added/removed with the mapview or simply just make the "useBuiltInZoomControls" as deprecated and remove it entirely

**Comment by monsieurtanuki** [comment]
I'm currently developing some kind of `ZoomButtonsOverlay`, inspired by what is in `SampleExtensive`.
Should be done by Sunday evening.

**Comment by monsieurtanuki** [comment]
Just pushed #741 
No more crash, as new class `ZoomButtonsOverlay` now replaces `ZoomButtonsController`.

**Comment by spyhunter99** [comment]
Probably need to add simpler layout options with the default being bottom center.
And the auto hide mechanism.

**Comment by spyhunter99** [comment]
ok i figured out how set the default to center bottom and added javadocs links to make it more obvious. Working on adding some padding now.
@MKergall any other suggestions?

**Comment by monsieurtanuki** [comment]
Fixed by #1190.
